src/test/java/RefactoringEngineTest.java

```diff
@@ -1,63 +1,59 @@
-import org.eclipse.jdt.core.dom.*;
-import org.junit.Test;
-import java.util.*;
-import static org.junit.Assert.*;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+import org.apache.commons.io.IOUtils;
+import org.eclipse.jdt.core.dom.AST;
+import org.eclipse.jdt.core.dom.ASTParser;
+import org.eclipse.jdt.core.dom.CompilationUnit;
+import org.eclipse.jdt.core.dom.Expression;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
 
 public class RefactoringEngineTest {
+    // Configure the refactoring engine
+    List<String> refactorings = Collections.singletonList("AddNullCheckBeforeDereferenceRefactoring");
+
+    @ParameterizedTest
+    @ValueSource(strings = { "TernaryBooleanFlagTest.java", "BooleanFlagTest.java", "NewContainerTest.java",
+            "SeperateVariableTest.java", "NestedNullCheck.java", "SentinelTest.java" })

```

Comment:
It would be better if we didn't need to specify each new test explicitly here, and instead this pointed at a folder containing all of the tests.